chain/borsh/decode_test.go

```diff
@@ -0,0 +1,313 @@
+package goborsh
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"testing"
+)
+
+func getDecoder(inputString string) BorshDecoder {
+	data, _ := hex.DecodeString(inputString)
+	return NewBorshDecoder(data)
+}
+
+func TestFinishedExpectFalse(t *testing.T) {
+	data, _ := hex.DecodeString("03000000425443")
+	decoder := NewBorshDecoder(data)
+	decodeFinished := decoder.Finished()
+
+	if decodeFinished {
+		t.Errorf(`Incorrect result for Finished bool: expected "%t" got "%t"`, false, decodeFinished)
```

Comment:
Our code base use [require](https://godoc.org/github.com/stretchr/testify/require) for testing